src/Adaptive.Agrona/IoUtil.cs

```diff
@@ -60,9 +60,9 @@ public static MemoryMappedFile OpenMemoryMappedFile(string path, MapMode mapMode
             var f = new FileStream(path, FileMode.Open, fileAccess, fileShare);
 
 #if NETFULL
-            return MemoryMappedFile.CreateFromFile(f, Guid.NewGuid().ToString(), 0, memoryMappedFileAccess, new MemoryMappedFileSecurity(), HandleInheritability.None, false);
+            return MemoryMappedFile.CreateFromFile(f, null, 0, memoryMappedFileAccess, new MemoryMappedFileSecurity(), HandleInheritability.None, false);
```

Comment:
Is there any documentation around this behaviour?

Reply:
Unfortunately it seems a bit hazy - my source was https://github.com/dotnet/corefx/issues/20091#issuecomment-303079998

But looking at the documentation here - https://msdn.microsoft.com/en-us/library/dn804422(v=vs.110).aspx - it says `ArgumentException when mapName is null or an empty string.`. 

However, in the official corefx tests, it has `null` being used https://github.com/dotnet/corefx/blob/master/src/System.IO.MemoryMappedFiles/tests/MemoryMappedFile.CrossProcess.cs#L13

See here for linux:

https://github.com/dotnet/corefx/blob/master/src/System.IO.MemoryMappedFiles/src/System/IO/MemoryMappedFiles/MemoryMappedFile.Unix.cs#L22-L34

But then the main class has this:

https://github.com/dotnet/corefx/blob/master/src/System.IO.MemoryMappedFiles/src/System/IO/MemoryMappedFiles/MemoryMappedFile.cs#L202-L205

Maybe this needs reverting.

Reply:
That last example is fine too - `(mapName != null && mapName.Length == 0)`.
So it's just the documentation that disagrees, happy to trust the code.